source/compiler-core/slang-source-loc.cpp

```diff
@@ -573,15 +573,38 @@ int SourceFile::calcColumnIndex(int lineIndex, int offset, int tabSize)
 
 void SourceFile::setContents(ISlangBlob* blob)
 {
-    const UInt contentSize = blob->getBufferSize();
+    const UInt rawContentSize = blob->getBufferSize();
 
-    SLANG_ASSERT(contentSize == m_contentSize);
+    SLANG_ASSERT(rawContentSize == m_contentSize);
 
-    char const* contentBegin = (char const*)blob->getBufferPointer();
-    char const* contentEnd = contentBegin + contentSize;
+    Byte* rawContentBegin = (Byte*)blob->getBufferPointer();
 
-    m_contentBlob = blob;
-    m_content = UnownedStringSlice(contentBegin, contentEnd);
+    // Query the encoding type and discard the Unicode Byte-Order-Marker before decoding
+    size_t offset;
+    auto type = CharEncoding::determineEncoding(
+        rawContentBegin,
+        rawContentSize,
+        offset);
+    SLANG_ASSERT(rawContentSize >= offset);
+
+    List<char> decodedBuffer;
+    CharEncoding::getEncoding(type)->decode(
+        rawContentBegin + offset,
+        int(rawContentSize - offset),
+        decodedBuffer);
+    String decodedBuffer2;
```

Comment:
If we are going to create a copy anyways, we can just do:
```
m_contentBlob = RawBlob::create(decidedBuffer.getBuffer(), decodedBuffer.getCount());
```

If we can live with some pointer punning UB (which we already do in many other places in this codebase anyways), we can avoid that copy with:

```
auto capacity = decodedBuffer.getCapacity();
auto count = decodedBuffer.getCount();
auto bufferPtr = (uint8_t*)decodedBuffer.detachBuffer();
List<uint8_t> u8buffer;
u8buffer.attachBuffer(bufferPtr, count, capacity);
m_contentBlob = ListBlob::moveCreate(u8buffer);
```

Reply:
What do you think of Jay's suggestion to use String and StringBlob?

Reply:
I don't know what the difference is in effect.

Reply:
Let's do either option I posted above. The reason for RawBlob is that it copies faster than per byte copies.

Reply:
Okay